backend/ee/danswer/external_permissions/slack/doc_sync.py

```diff
@@ -0,0 +1,187 @@
+from slack_sdk import WebClient
+from sqlalchemy.orm import Session
+
+from danswer.access.models import ExternalAccess
+from danswer.connectors.factory import instantiate_connector
+from danswer.connectors.interfaces import IdConnector
+from danswer.connectors.models import InputType
+from danswer.connectors.slack.connector import get_channels
+from danswer.connectors.slack.connector import make_paginated_slack_api_call_w_retries
+from danswer.db.models import ConnectorCredentialPair
+from danswer.db.users import batch_add_non_web_user_if_not_exists__no_commit
+from danswer.utils.logger import setup_logger
+from ee.danswer.db.document import upsert_document_external_perms__no_commit
+from ee.danswer.external_permissions.slack.utils import fetch_user_id_to_email_map
+
+
+logger = setup_logger()
+
+
+def _extract_channel_id_from_doc_id(doc_id: str) -> str:
+    """
+    Extracts the channel ID from a document ID string.
+
+    The document ID is expected to be in the format: "{channel_id}__{message_ts}"
+
+    Args:
+        doc_id (str): The document ID string.
+
+    Returns:
+        str: The extracted channel ID.
+
+    Raises:
+        ValueError: If the doc_id doesn't contain the expected separator.
+    """
+    try:
+        channel_id, _ = doc_id.split("__", 1)
```

Comment:
Try not to have this type of magical logic. Ideally if we can put the channel in the doc metadata and pull it from there and throw an exception if it's not found or something. That would be ideal (this can be a backoff check for messages that don't have a channel in the metadata)

Reply:
I agree, that is why i commented thouroughly. The problem is the ID only connector. But maybe we shoudl have a version of the connector that retrieves id+metadata? probably worth a discussion. Regardless, the rework wouldnt be that difficult if we delay the decision